src/core/src/bootstrap/EnvLayer.py

```diff
@@ -64,20 +64,25 @@ def get_package_manager(self):
         """ Detects package manager type """
         ret = None
 
-        # choose default - almost surely one will match.
-        for b in ('apt-get', 'yum', 'zypper'):
-            code, out = self.run_command_output('which ' + b, False, False)
+        if self.platform.linux_distribution()[0] == Constants.AZURE_LINUX:
```

Comment:
If we want to support any other OS with tdnf package manager, we need to make change again and add this in if condition. any better way to do this without checking for OS but just check for "which tdnf"?

Reply:
We are only adding support for this because Az Linux 3 requires it. We do not want to introduce unwanted support matrix changes when this code goes out. The scope is deliberately limited for safety. Expansion will similarly be covered much later when the ROI dictates the one-time and perpetual cost is worth bearing.